tests/components/vacuum/test_init.py

```diff
@@ -0,0 +1,18 @@
+"""The tests for Vacuum."""
+from homeassistant.components import vacuum
+
+
+def test_deprecated_base_class(caplog):
+    """Test deprecated base class."""
+
+    class CustomVacuum(vacuum.VacuumDevice):
+        pass
+
+    class CustomStateVacuum(vacuum.StateVacuumDevice):
+        pass
+
+    CustomVacuum()
+    assert "VacuumDevice is deprecated, modify CustomVacuum" in caplog.text
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5418277)